keps/sig-node/4800-cpumanager-split-l3cache/README.md

```diff
@@ -0,0 +1,1235 @@
+
+![Figure to the results of l3 cache alignment](https://github.com/ajcaldelas/enhancements/blob/c7e423439608001cc837e2f5629dc87935353611/keps/sig-node/4800-cpumanager-split-l3cache/images/results.png) ![Figure of results](https://github.com/ajcaldelas/enhancements/blob/c7e423439608001cc837e2f5629dc87935353611/keps/sig-node/4800-cpumanager-split-l3cache/images/chart.png)
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+- Introduce a new CPU Manager policy option that assigns CPU within the same grouping of uncorecache or L3 cache to pods and container scope.  
+- Minimize the number of CPUs assignments from different uncorecache/L3 grouping to pods and container scope.
+- Ensure the uncore cache alignment feature works with the Topology Manager Policies to ensure alignment based on NUMA and device hints. 
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+
+This proposal does not aim to modify CPU assignments for CPU Manager policy set to none.  This does not alter the behavior of existing static policy options such as full-pcpus-only.
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+- Add a new static policy option to fine-tune of CPU assignments to align by uncorecache/L3 cache grouping.
+
+- Topology struct already contains CPUDetails which is a map of CPUInfo.  CPUInfo knows about NUMA, socket, and core IDs associated with a CPU.  We just need to add a new member called uncorecacheId that tracks whether the CPU is part of a split L3. Functionality to support this was merged in cadvisor with these pull-requests [cadvisor/pr-2849](https://github.com/google/cadvisor/pull/2849) and [cadvisor/pr-2847](https://github.com/google/cadvisor/pull/2847/)
+
+- Handle enablement of the policy option in pkg/kubelet/cm/cpumanager/policy_options.go and check the validity of the user flag and capability of the system.  If topology does not support split L3, grouping by L3 static policy will be ignored.  
+
+- Modify the "Allocate" static policy to check for the option, align-cpus-by-uncorecache. For platforms where SMT is enabled, align-cpus-by-uncorecache will typically take full cores and also works with the option full-pcpus-only.
```

Comment:
The dependency to check for the align-by-uncore-cache option is because we still want the default static policy as well. There are cases where users won't necessarily need the uncore cache alignment.
We are creating this feature for the default packed core/NUMA setting (narrowing the scope). We plan for future support, but the distribution across cores is typically beneficial for low node utilization, which is less common and lower priority.

Reply:
taking full cores when possible is the standard behavior since the inception of cpumanager, and it's only enforced (vs best effort) when `full-pcpus-only` is in use. We should probably rephrasing saying that besides uncore cache alignment the allocation code will keep behaving as before, for example trying to allocate full cores when possibler.

Reply:
Updated kep with your feedback